services/src/main/java/se/inera/intyg/common/services/messages/DefaultCertificateMessagesProvider.java

```diff
@@ -23,18 +23,32 @@
 public final class DefaultCertificateMessagesProvider implements CertificateMessagesProvider {
 
     private final Map<String, String> messages;
+    private Map<String, String> dynamicLabels;
 
-    private DefaultCertificateMessagesProvider(Map<String, String> messages) {
-        this.messages = messages;
+    public static CertificateMessagesProvider create(Map<String, String> messages, Map<String, String> dynamicLabels) {
+        return new DefaultCertificateMessagesProvider(messages, dynamicLabels);
     }
 
     public static CertificateMessagesProvider create(Map<String, String> messages) {
         return new DefaultCertificateMessagesProvider(messages);
     }
 
+    private DefaultCertificateMessagesProvider(Map<String, String> messages, Map<String, String> dynamicLabels) {
+        this.messages = messages;
+        this.dynamicLabels = dynamicLabels;
+    }
+
+    private DefaultCertificateMessagesProvider(Map<String, String> messages) {
+        this.messages = messages;
```

Comment:
Suggest you  remove this constructor and only keep the one for messages and dynamicLabels. Then the factory method that only takes messages can create an empty Map and use it for the constructor. This will also reduce the risk of NullPointerExceptions.